src/main/java/core/basesyntax/service/impl/ParserServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ParserService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class ParserServiceImpl implements ParserService {
+    private static final int CODE_POSITION = 0;
+    private static final int FRUIT_POSITION = 1;
+    private static final int QUANTITY_POSITION = 2;
+    private static final String SEPARATOR = ",";
+    private static final int PARTS_NUMBER = 3;
+
+    @Override
+    public List<FruitTransaction> parse(List<String> lines) {
+        List<FruitTransaction> fruitTransactionList = new ArrayList<>();
+        boolean firstLine = true;
+        for (String line : lines) {
+            if (firstLine) {
+                firstLine = false;
+                continue;
+            }
+            String[] parts = line.split(SEPARATOR);
+            if (parts.length != PARTS_NUMBER) {
+                continue;
+            }
+            Optional<FruitTransaction.Operation> operation =
+                    FruitTransaction.Operation.fromString(parts[CODE_POSITION]);
+            if (operation.isEmpty()) {
+                continue;
+            }
+            String fruit = parts[FRUIT_POSITION];
+            int quantity;
+            try {
+                quantity = Integer.parseInt(parts[QUANTITY_POSITION]);
+            } catch (NumberFormatException e) {
+                continue;
```

Comment:
Catching `NumberFormatException` without any feedback to the user can lead to hard-to-find bugs if the input data is incorrect. It would be better to throw a custom exception with a clear message indicating what went wrong.